### ISSUE TITLE
Reviewing for JOSS: Increase explanations for POSTGRES_PASSWORD .env

### ISSUE THREAD

**Issue by LukeParky** [GeospatialResearch/Digital-Twins#342]
This issue comes from the JOSS review openjournals/joss-reviews#7433

> > Hi @pritamd47, for your review items below, could you please expand on these? Have you encountered errors, or is there another barrier to installation that impacts review?
> > > ### Functionality
> > > 
> > > * [ ]    **Installation:** Does installation proceed as outlined in the documentation?
> > > * [ ]    **Functionality:** Have the functional claims of the software been confirmed?
> 
> Hi @LukeParky, I was following older version of the documentation on installing the software due to which I couldn't install it. earlier. I see that you have updated the documentation since. I tried installing the software again and now I think I am getting an error because the "POSTGRES_PASSWORD" field in my .env file is empty. How do I get this password?

It looks like the documntation can be clarified.

**Comment by pritamd47** [comment]
Hi @LukeParky, thanks for creating the issue. I was able to resolve this issue. However, I failed to run it on my computer which is mac (m1 chip) and got an error of mismatch in architecture when I tried to run docker. I was able to run it on another machine using WSL (linux emulation for windows). However, I was still getting some errors. Please take a look at the attached screenshots.

As suggested by documentation, when I visit 8080 port, I get the following error - 
![image (1)](https://github.com/user-attachments/assets/6b815343-ee0f-41bd-9ac7-1c63982e2f5b)

From the docker GUI I opened another link to a port that seemed to load the tool but there I got the following error - 
![image](https://github.com/user-attachments/assets/378c305f-50c7-4761-8c9e-be3073e089fe)

Could you help me with these errors while installing the software?

**Comment by LukeParky** [comment]
Hi @pritamd47, thank you for making aware of the issue about mac, this may be able to be resolved by building the image manually but I do not have access to a mac to test.
I will provide an update to the documentation for the web port.

As for the error message you are receiving, could you please provide your logs with `docker compose logs -f backend celery_worker`?
Do you still receive the error message?

**Comment by LukeParky** [comment]
You may run into the issue that the other reviewer is encountering, https://github.com/GeospatialResearch/Digital-Twins/issues/344, working on a fix

**Comment by pritamd47** [comment]
I lost access to the WSL machine I was working with earlier, so trying to setup the server in a new machine. I will keep you updated.

**Comment by LukeParky** [comment]
Hi @pritamd47 I recommend git pulling from the newest changes, and then pulling/building a new docker image.
I am not aware of any further issues that should impact your review but please reach out if you are having trouble.

**Comment by pritamd47** [comment]
Hi Luke. I pulled the latest changes as you instructed and started ran the `docker compose up -d` command. The container seems to be running. 

![Image](https://github.com/user-attachments/assets/93c3ddec-6549-4eb8-8c6c-9e568da740d9)

However, localhost:8080 throws a connection refused error. The celery_logs are attached below.
```
backend_digital_twin        | [2025-01-24 07:18:49 +0000] [1] [INFO] Starting gunicorn 20.1.0
backend_digital_twin        | [2025-01-24 07:18:49 +0000] [1] [INFO] Listening at: http://0.0.0.0:5000 (1)
backend_digital_twin        | [2025-01-24 07:18:49 +0000] [1] [INFO] Using worker: sync
backend_digital_twin        | [2025-01-24 07:18:49 +0000] [13] [INFO] Booting worker with pid: 13
backend_digital_twin        | /venv/lib/python3.11/site-packages/geopandas/_compat.py:123: UserWarning: The Shapely GEOS version (3.11.2-CAPI-1.17.2) is incompatible with the GEOS version PyGEOS was compiled with (3.10.4-CAPI-1.16.2). Conversions between both will be slow.
backend_digital_twin        |   warnings.warn(
backend_digital_twin        | /app/src/tasks.py:25: UserWarning: Shapely 2.0 is installed, but because PyGEOS is also installed, GeoPandas will still use PyGEOS by default for now. To force to use and test Shapely 2.0, you have to set the environment variable USE_PYGEOS=0. You can do this before starting the Python process, or in your code before importing geopandas:
backend_digital_twin        | 
backend_digital_twin        | import os
backend_digital_twin        | os.environ['USE_PYGEOS'] = '0'
backend_digital_twin        | import geopandas
backend_digital_twin        | 
backend_digital_twin        | In a future release, GeoPandas will switch to using Shapely by default. If you are using PyGEOS directly (calling PyGEOS functions on geometries from GeoPandas), this will then stop working and you are encouraged to migrate from PyGEOS to Shapely 2.0 (https://shapely.readthedocs.io/en/latest/migration_pygeos.html).
backend_digital_twin        |   import geopandas as gpd
celery_worker_digital_twin  | /venv/lib/python3.11/site-packages/geopandas/_compat.py:123: UserWarning: The Shapely GEOS version (3.11.2-CAPI-1.17.2) is incompatible with the GEOS version PyGEOS was compiled with (3.10.4-CAPI-1.16.2). Conversions between both will be slow.
celery_worker_digital_twin  |   warnings.warn(
celery_worker_digital_twin  | /app/src/tasks.py:25: UserWarning: Shapely 2.0 is installed, but because PyGEOS is also installed, GeoPandas will still use PyGEOS by default for now. To force to use and test Shapely 2.0, you have to set the environment variable USE_PYGEOS=0. You can do this before starting the Python process, or in your code before importing geopandas:
celery_worker_digital_twin  | 
celery_worker_digital_twin  | import os
celery_worker_digital_twin  | os.environ['USE_PYGEOS'] = '0'
celery_worker_digital_twin  | import geopandas
celery_worker_digital_twin  | 
celery_worker_digital_twin  | In a future release, GeoPandas will switch to using Shapely by default. If you are using PyGEOS directly (calling PyGEOS functions on geometries from GeoPandas), this will then stop working and you are encouraged to migrate from PyGEOS to Shapely 2.0 (https://shapely.readthedocs.io/en/latest/migration_pygeos.html).
celery_worker_digital_twin  |   import geopandas as gpd
celery_worker_digital_twin  |  
celery_worker_digital_twin  |  -------------- celery@719d983c4d65 v5.2.7 (dawn-chorus)
celery_worker_digital_twin  | --- ***** ----- 
celery_worker_digital_twin  | -- ******* ---- Linux-5.15.153.1-microsoft-standard-WSL2-x86_64-with-glibc2.35 2025-01-24 07:18:53
celery_worker_digital_twin  | - *** --- * --- 
celery_worker_digital_twin  | - ** ---------- [config]
celery_worker_digital_twin  | - ** ---------- .> app:         tasks:0x7f5773591dd0
celery_worker_digital_twin  | - ** ---------- .> transport:   redis://message_broker:6379/0
celery_worker_digital_twin  | - ** ---------- .> results:     redis://message_broker:6379/0
celery_worker_digital_twin  | - *** --- * --- .> concurrency: 16 (thread)
celery_worker_digital_twin  | -- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
celery_worker_digital_twin  | --- ***** ----- 
celery_worker_digital_twin  |  -------------- [queues]
celery_worker_digital_twin  |                 .> celery           exchange=celery(direct) key=celery
celery_worker_digital_twin  |                 
celery_worker_digital_twin  | 
celery_worker_digital_twin  | [tasks]
celery_worker_digital_twin  |   . src.tasks.add_base_data_to_db
celery_worker_digital_twin  |   . src.tasks.generate_rainfall_inputs
celery_worker_digital_twin  |   . src.tasks.generate_river_inputs
celery_worker_digital_twin  |   . src.tasks.generate_tide_inputs
celery_worker_digital_twin  |   . src.tasks.get_depth_by_time_at_point
celery_worker_digital_twin  |   . src.tasks.get_model_extents_bbox
celery_worker_digital_twin  |   . src.tasks.get_model_output_filepath_from_model_id
celery_worker_digital_twin  |   . src.tasks.get_valid_parameters_based_on_confidence_level
celery_worker_digital_twin  |   . src.tasks.process_dem
celery_worker_digital_twin  |   . src.tasks.refresh_lidar_datasets
celery_worker_digital_twin  |   . src.tasks.run_flood_model
celery_worker_digital_twin  |   . src.tasks.validate_slr_parameters
celery_worker_digital_twin  | 
celery_worker_digital_twin  | (PDAL Error) Can't load library /venv/lib/libpdal_plugin_reader_draco.so: Failed to load "/venv/lib/libpdal_plugin_reader_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_writer_draco.so: Failed to load "/venv/lib/libpdal_plugin_writer_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_reader_draco.so: Failed to load "/venv/lib/libpdal_plugin_reader_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_writer_draco.so: Failed to load "/venv/lib/libpdal_plugin_writer_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory[2025-01-24 07:18:53,428: INFO/MainProcess] Connected to redis://message_broker:6379/0
celery_worker_digital_twin  | [2025-01-24 07:18:53,430: INFO/MainProcess] mingle: searching for neighbors
celery_worker_digital_twin  | [2025-01-24 07:18:54,438: INFO/MainProcess] mingle: all alone
celery_worker_digital_twin  | [2025-01-24 07:18:54,460: INFO/MainProcess] celery@719d983c4d65 ready.
backend_digital_twin        | (PDAL Error) Can't load library /venv/lib/libpdal_plugin_reader_draco.so: Failed to load "/venv/lib/libpdal_plugin_reader_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_writer_draco.so: Failed to load "/venv/lib/libpdal_plugin_writer_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_reader_draco.so: Failed to load "/venv/lib/libpdal_plugin_reader_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_writer_draco.so: Failed to load "/venv/lib/libpdal_plugin_writer_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory[2025-01-24 07:22:10 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:13)
backend_digital_twin        | [2025-01-24 07:22:10 +0000] [13] [INFO] Worker exiting (pid: 13)
backend_digital_twin        | [2025-01-24 07:22:10 +0000] [160] [INFO] Booting worker with pid: 160
backend_digital_twin        | /venv/lib/python3.11/site-packages/geopandas/_compat.py:123: UserWarning: The Shapely GEOS version (3.11.2-CAPI-1.17.2) is incompatible with the GEOS version PyGEOS was compiled with (3.10.4-CAPI-1.16.2). Conversions between both will be slow.
backend_digital_twin        |   warnings.warn(
backend_digital_twin        | /app/src/tasks.py:25: UserWarning: Shapely 2.0 is installed, but because PyGEOS is also installed, GeoPandas will still use PyGEOS by default for now. To force to use and test Shapely 2.0, you have to set the environment variable USE_PYGEOS=0. You can do this before starting the Python process, or in your code before importing geopandas:
backend_digital_twin        | 
backend_digital_twin        | import os
backend_digital_twin        | os.environ['USE_PYGEOS'] = '0'
backend_digital_twin        | import geopandas
backend_digital_twin        | 
backend_digital_twin        | In a future release, GeoPandas will switch to using Shapely by default. If you are using PyGEOS directly (calling PyGEOS functions on geometries from GeoPandas), this will then stop working and you are encouraged to migrate from PyGEOS to Shapely 2.0 (https://shapely.readthedocs.io/en/latest/migration_pygeos.html).
backend_digital_twin        |   import geopandas as gpd

```

**Comment by LukeParky** [comment]
Hi @pritamd47, if localhost:8080 then perhaps your port does not match. Could you please provide the output of `docker compose ps`? It should tell you what port to use.

If that is not the issue, could you please take a screenshot of the connection refused error?
It may be good for me to provide some one-on-one support to make sure this review can be completed, and see if there are ways we can improve the setup instructions, which time-zone are you in?

**Comment by pritamd47** [comment]
Hi @LukeParky, sorry for the delay in getting back. Last couple of weeks were busy for me as I will be defending my dissertation soon :)

**Comment by pritamd47** [comment]
Hi @LukeParky, sorry about the delay in getting back. I was quite busy this past couple of weeks because I have my phd defense coming up.

I have I tried out the `docker compose ps` command, and you're right that the port doesn't quite match up. The `www` service is running on 5173 port for some reason. Here is a screenshot of the output:
![Image](https://github.com/user-attachments/assets/f5681eee-cf5d-498e-9d90-1a49c5e544b0)

When I visit this port at `http://localhost:5173/`, I get the following screen:

![Image](https://github.com/user-attachments/assets/aaf595a9-45e4-4f99-8862-a729c5121611)

Here are the celery_logs:
```
backend_digital_twin        | [2025-03-01 06:49:32 +0000] [1] [INFO] Starting gunicorn 20.1.0
backend_digital_twin        | [2025-03-01 06:49:32 +0000] [1] [INFO] Listening at: http://0.0.0.0:5000 (1)
backend_digital_twin        | [2025-03-01 06:49:32 +0000] [1] [INFO] Using worker: sync
backend_digital_twin        | [2025-03-01 06:49:32 +0000] [14] [INFO] Booting worker with pid: 14
backend_digital_twin        | /venv/lib/python3.11/site-packages/geopandas/_compat.py:123: UserWarning: The Shapely GEOS version (3.11.2-CAPI-1.17.2) is incompatible with the GEOS version PyGEOS was compiled with (3.10.4-CAPI-1.16.2). Conversions between both will be slow.
backend_digital_twin        |   warnings.warn(
backend_digital_twin        | /app/src/tasks.py:25: UserWarning: Shapely 2.0 is installed, but because PyGEOS is also installed, GeoPandas will still use PyGEOS by default for now. To force to use and test Shapely 2.0, you have to set the environment variable USE_PYGEOS=0. You can do this before starting the Python process, or in your code before importing geopandas:
backend_digital_twin        | 
backend_digital_twin        | import os
backend_digital_twin        | os.environ['USE_PYGEOS'] = '0'
backend_digital_twin        | import geopandas
backend_digital_twin        | 
backend_digital_twin        | In a future release, GeoPandas will switch to using Shapely by default. If you are using PyGEOS directly (calling PyGEOS functions on geometries from GeoPandas), this will then stop working and you are encouraged to migrate from PyGEOS to Shapely 2.0 (https://shapely.readthedocs.io/en/latest/migration_pygeos.html).
backend_digital_twin        |   import geopandas as gpd
celery_worker_digital_twin  | /venv/lib/python3.11/site-packages/geopandas/_compat.py:123: UserWarning: The Shapely GEOS version (3.11.2-CAPI-1.17.2) is incompatible with the GEOS version PyGEOS was compiled with (3.10.4-CAPI-1.16.2). Conversions between both will be slow.
celery_worker_digital_twin  |   warnings.warn(
celery_worker_digital_twin  | /app/src/tasks.py:25: UserWarning: Shapely 2.0 is installed, but because PyGEOS is also installed, GeoPandas will still use PyGEOS by default for now. To force to use and test Shapely 2.0, you have to set the environment variable USE_PYGEOS=0. You can do this before starting the Python process, or in your code before importing geopandas:
celery_worker_digital_twin  | 
celery_worker_digital_twin  | import os
celery_worker_digital_twin  | os.environ['USE_PYGEOS'] = '0'
celery_worker_digital_twin  | import geopandas
celery_worker_digital_twin  | 
celery_worker_digital_twin  | In a future release, GeoPandas will switch to using Shapely by default. If you are using PyGEOS directly (calling PyGEOS functions on geometries from GeoPandas), this will then stop working and you are encouraged to migrate from PyGEOS to Shapely 2.0 (https://shapely.readthedocs.io/en/latest/migration_pygeos.html).
celery_worker_digital_twin  |   import geopandas as gpd
celery_worker_digital_twin  |  
celery_worker_digital_twin  |  -------------- celery@6db85fb2c6cf v5.2.7 (dawn-chorus)
celery_worker_digital_twin  | --- ***** ----- 
celery_worker_digital_twin  | -- ******* ---- Linux-5.15.153.1-microsoft-standard-WSL2-x86_64-with-glibc2.35 2025-03-01 06:49:37
celery_worker_digital_twin  | - *** --- * --- 
celery_worker_digital_twin  | - ** ---------- [config]
celery_worker_digital_twin  | - ** ---------- .> app:         tasks:0x7f7b9243dd90
celery_worker_digital_twin  | - ** ---------- .> transport:   redis://message_broker:6379/0
celery_worker_digital_twin  | - ** ---------- .> results:     redis://message_broker:6379/0
celery_worker_digital_twin  | - *** --- * --- .> concurrency: 16 (thread)
celery_worker_digital_twin  | -- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
celery_worker_digital_twin  | --- ***** ----- 
celery_worker_digital_twin  |  -------------- [queues]
celery_worker_digital_twin  |                 .> celery           exchange=celery(direct) key=celery
celery_worker_digital_twin  |                 
celery_worker_digital_twin  | 
celery_worker_digital_twin  | [tasks]
celery_worker_digital_twin  |   . src.tasks.add_base_data_to_db
celery_worker_digital_twin  |   . src.tasks.generate_rainfall_inputs
celery_worker_digital_twin  |   . src.tasks.generate_river_inputs
celery_worker_digital_twin  |   . src.tasks.generate_tide_inputs
celery_worker_digital_twin  |   . src.tasks.get_depth_by_time_at_point
celery_worker_digital_twin  |   . src.tasks.get_model_extents_bbox
celery_worker_digital_twin  |   . src.tasks.get_model_output_filepath_from_model_id
celery_worker_digital_twin  |   . src.tasks.get_valid_parameters_based_on_confidence_level
celery_worker_digital_twin  |   . src.tasks.process_dem
celery_worker_digital_twin  |   . src.tasks.refresh_lidar_datasets
celery_worker_digital_twin  |   . src.tasks.run_flood_model
celery_worker_digital_twin  |   . src.tasks.validate_slr_parameters
celery_worker_digital_twin  | 
celery_worker_digital_twin  | (PDAL Error) Can't load library /venv/lib/libpdal_plugin_reader_draco.so: Failed to load "/venv/lib/libpdal_plugin_reader_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_writer_draco.so: Failed to load "/venv/lib/libpdal_plugin_writer_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_reader_draco.so: Failed to load "/venv/lib/libpdal_plugin_reader_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_writer_draco.so: Failed to load "/venv/lib/libpdal_plugin_writer_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory[2025-03-01 06:49:37,420: INFO/MainProcess] Connected to redis://message_broker:6379/0
celery_worker_digital_twin  | [2025-03-01 06:49:37,427: INFO/MainProcess] mingle: searching for neighbors
celery_worker_digital_twin  | [2025-03-01 06:49:38,436: INFO/MainProcess] mingle: all alone
celery_worker_digital_twin  | [2025-03-01 06:49:38,450: INFO/MainProcess] celery@6db85fb2c6cf ready.
celery_worker_digital_twin  | [2025-03-01 06:50:27,073: INFO/MainProcess] Task src.tasks.get_valid_parameters_based_on_confidence_level[a03fca60-e5ad-4d80-b830-c6ebc462dd4f] received
celery_worker_digital_twin  | [2025-03-01 06:50:27,148: INFO/MainProcess] Fetching 'sea_level_rise' data from NZ SeaRise Takiwa.
backend_digital_twin        | (PDAL Error) Can't load library /venv/lib/libpdal_plugin_reader_draco.so: Failed to load "/venv/lib/libpdal_plugin_reader_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_writer_draco.so: Failed to load "/venv/lib/libpdal_plugin_writer_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_reader_draco.so: Failed to load "/venv/lib/libpdal_plugin_reader_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_writer_draco.so: Failed to load "/venv/lib/libpdal_plugin_writer_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory[2025-03-01 06:50:56 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:14)
backend_digital_twin        | [2025-03-01 06:50:56 +0000] [14] [INFO] Worker exiting (pid: 14)
backend_digital_twin        | [2025-03-01 06:50:57 +0000] [83] [INFO] Booting worker with pid: 83
backend_digital_twin        | /venv/lib/python3.11/site-packages/geopandas/_compat.py:123: UserWarning: The Shapely GEOS version (3.11.2-CAPI-1.17.2) is incompatible with the GEOS version PyGEOS was compiled with (3.10.4-CAPI-1.16.2). Conversions between both will be slow.
backend_digital_twin        |   warnings.warn(
backend_digital_twin        | /app/src/tasks.py:25: UserWarning: Shapely 2.0 is installed, but because PyGEOS is also installed, GeoPandas will still use PyGEOS by default for now. To force to use and test Shapely 2.0, you have to set the environment variable USE_PYGEOS=0. You can do this before starting the Python process, or in your code before importing geopandas:
backend_digital_twin        | 
backend_digital_twin        | import os
backend_digital_twin        | os.environ['USE_PYGEOS'] = '0'
backend_digital_twin        | import geopandas
backend_digital_twin        | 
backend_digital_twin        | In a future release, GeoPandas will switch to using Shapely by default. If you are using PyGEOS directly (calling PyGEOS functions on geometries from GeoPandas), this will then stop working and you are encouraged to migrate from PyGEOS to Shapely 2.0 (https://shapely.readthedocs.io/en/latest/migration_pygeos.html).
backend_digital_twin        |   import geopandas as gpd
celery_worker_digital_twin  | [2025-03-01 06:51:01,988: INFO/MainProcess] Task src.tasks.get_valid_parameters_based_on_confidence_level[ef98c363-49a6-4d46-9ad0-64d10ad567e7] received
celery_worker_digital_twin  | [2025-03-01 06:51:02,015: INFO/MainProcess] Fetching 'sea_level_rise' data from NZ SeaRise Takiwa.
backend_digital_twin        | (PDAL Error) Can't load library /venv/lib/libpdal_plugin_reader_draco.so: Failed to load "/venv/lib/libpdal_plugin_reader_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_writer_draco.so: Failed to load "/venv/lib/libpdal_plugin_writer_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_reader_draco.so: Failed to load "/venv/lib/libpdal_plugin_reader_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory(PDAL Error) Can't load library /venv/lib/libpdal_plugin_writer_draco.so: Failed to load "/venv/lib/libpdal_plugin_writer_draco.so": libdraco.so.8: cannot open shared object file: No such file or directory[2025-03-01 06:51:30 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:83)
backend_digital_twin        | [2025-03-01 06:51:30 +0000] [83] [INFO] Worker exiting (pid: 83)
backend_digital_twin        | [2025-03-01 06:51:31 +0000] [114] [INFO] Booting worker with pid: 114
backend_digital_twin        | /venv/lib/python3.11/site-packages/geopandas/_compat.py:123: UserWarning: The Shapely GEOS version (3.11.2-CAPI-1.17.2) is incompatible with the GEOS version PyGEOS was compiled with (3.10.4-CAPI-1.16.2). Conversions between both will be slow.
backend_digital_twin        |   warnings.warn(
backend_digital_twin        | /app/src/tasks.py:25: UserWarning: Shapely 2.0 is installed, but because PyGEOS is also installed, GeoPandas will still use PyGEOS by default for now. To force to use and test Shapely 2.0, you have to set the environment variable USE_PYGEOS=0. You can do this before starting the Python process, or in your code before importing geopandas:
backend_digital_twin        | 
backend_digital_twin        | import os
backend_digital_twin        | os.environ['USE_PYGEOS'] = '0'
backend_digital_twin        | import geopandas
backend_digital_twin        | 
backend_digital_twin        | In a future release, GeoPandas will switch to using Shapely by default. If you are using PyGEOS directly (calling PyGEOS functions on geometries from GeoPandas), this will then stop working and you are encouraged to migrate from PyGEOS to Shapely 2.0 (https://shapely.readthedocs.io/en/latest/migration_pygeos.html).
backend_digital_twin        |   import geopandas as gpd
celery_worker_digital_twin  | [2025-03-01 06:51:36,647: INFO/MainProcess] Successfully fetched the 'NZSeaRise_proj_novlm.csv' data.
celery_worker_digital_twin  | [2025-03-01 06:51:41,602: INFO/MainProcess] Task src.tasks.get_valid_parameters_based_on_confidence_level[54d6f759-9fce-48e1-b7c7-4556ec1a1529] received
celery_worker_digital_twin  | [2025-03-01 06:51:41,639: INFO/MainProcess] Fetching 'sea_level_rise' data from NZ SeaRise Takiwa.
```

**Comment by pritamd47** [comment]
I am in Pacific Standard Time (PST). I'd be happy to get on a call to see what might be going wrong at my end.

**Comment by LukeParky** [comment]
Hi @pritamd47, thank you for getting back to me. I understand this will be a very busy time for you so I think having a quick call might be the best way to make sure your time for this review is efficient.

I think that all you need to do is refresh your browser window, unless another issue is coming up further down the logs.

I'll check if a call like this is allowed with the editor, but in the meantime please let me know if refreshing the browser works.

**Comment by pritamd47** [comment]
Hi Luke,

I actually tried refreshing my window after seeing the error but got the
same error. Should I wait for a while before hitting refresh?

I’ll try again sometime today. If it doesn’t work, a call would be great to
see what could be going wrong and if it is something local to the system I
am using.

Pritam

**Comment by LukeParky** [comment]
Hi @pritamd47, I have sent a Teams meeting invite to the email address on your github profile. Please let me know if it's suitable.

**Comment by pritamd47** [comment]
Hi @LukeParky, thank you for getting on call to help me with setting up of the software. I am getting the same error of `PermissionError: [Error 13] Permission denied: '/stored_data/lidar'`. Here are the full logs - 

```
celery_worker_digital_twin  | [2025-03-05 05:02:39,295: INFO/MainProcess] dataset table does not exist, initialising LiDAR dataset information.
celery_worker_digital_twin  | [2025-03-05 05:02:39,295: INFO/MainProcess] Start crawling datasets from OpenTopography.
celery_worker_digital_twin  | [2025-03-05 05:02:49,562: DEBUG/MainProcess] pidbox received method ping() [reply_to:{'exchange': 'reply.celery.pidbox', 'routing_key': 'e75ee036-8df9-34bd-88c6-f130c1b3bf2a'} ticket:65385dc9-5b1d-465c-9d22-d69c61deaf15]
celery_worker_digital_twin  | [2025-03-05 05:03:04,061: DEBUG/MainProcess] pidbox received method ping() [reply_to:{'exchange': 'reply.celery.pidbox', 'routing_key': 'dfe85883-c58c-33d7-8c43-f25169048daa'} ticket:17a8a07a-3d05-446d-80c4-c43f8f9b72b2]
celery_worker_digital_twin  | [2025-03-05 05:03:18,931: DEBUG/MainProcess] pidbox received method ping() [reply_to:{'exchange': 'reply.celery.pidbox', 'routing_key': 'da409363-13e2-36d3-ba2d-0dcb9ec1597c'} ticket:bf49907d-be82-4855-b0a4-26b34b34e54e]
celery_worker_digital_twin  | [2025-03-05 05:03:34,527: DEBUG/MainProcess] pidbox received method ping() [reply_to:{'exchange': 'reply.celery.pidbox', 'routing_key': '8b25fb17-1e8e-344c-bcd5-115be11f6f3d'} ticket:ce1a6902-6306-46dd-9b25-f63b91155f88]
celery_worker_digital_twin  | [2025-03-05 05:03:49,423: DEBUG/MainProcess] pidbox received method ping() [reply_to:{'exchange': 'reply.celery.pidbox', 'routing_key': '1396396d-5a95-338f-b053-8ff8b4f06cdc'} ticket:23e55ab2-2fd7-4d34-af12-307dd6e6e3ec]
celery_worker_digital_twin  | [2025-03-05 05:04:04,704: DEBUG/MainProcess] pidbox received method ping() [reply_to:{'exchange': 'reply.celery.pidbox', 'routing_key': 'ea180508-a82a-306a-a44e-0618cfe6cc48'} ticket:51274d6f-e002-44fb-a503-058976455d69]
celery_worker_digital_twin  | [2025-03-05 05:04:19,780: DEBUG/MainProcess] pidbox received method ping() [reply_to:{'exchange': 'reply.celery.pidbox', 'routing_key': '30c3acb5-834c-346d-96e5-a33230076a5a'} ticket:7a2d5a68-5f35-4203-89f1-dd52aa342249]
celery_worker_digital_twin  | [2025-03-05 05:04:34,740: DEBUG/MainProcess] pidbox received method ping() [reply_to:{'exchange': 'reply.celery.pidbox', 'routing_key': '69cd6299-40fe-3108-a8d8-04c39ae8e62e'} ticket:c6af476f-9a59-4c9a-8916-cdd91c85e0bc]
celery_worker_digital_twin  | [2025-03-05 05:04:49,600: DEBUG/MainProcess] pidbox received method ping() [reply_to:{'exchange': 'reply.celery.pidbox', 'routing_key': 'd2dbf587-d214-36a3-a317-55ec20ffe720'} ticket:77c49fa3-f4e8-4dca-aee2-5366169d14a5]
celery_worker_digital_twin  | [2025-03-05 05:05:04,502: DEBUG/MainProcess] pidbox received method ping() [reply_to:{'exchange': 'reply.celery.pidbox', 'routing_key': 'e66bf8b3-b0d7-3b7e-a90c-e80a39145b82'} ticket:9cd3369a-9cc0-41c2-9f0c-9161f7fe799a]
celery_worker_digital_twin  | [2025-03-05 05:05:19,824: DEBUG/MainProcess] pidbox received method ping() [reply_to:{'exchange': 'reply.celery.pidbox', 'routing_key': '9665ad6c-97c7-371b-9559-98eb4e4869ae'} ticket:6c7ac1a4-eb63-4b39-8d5f-372e60a5b039]
celery_worker_digital_twin  | [2025-03-05 05:05:34,842: DEBUG/MainProcess] pidbox received method ping() [reply_to:{'exchange': 'reply.celery.pidbox', 'routing_key': '34c414b9-b064-34eb-b698-3c83e133b1af'} ticket:223afcea-4526-4456-acc2-6b7559ef48cb]
celery_worker_digital_twin  | [2025-03-05 05:05:39,460: INFO/MainProcess] Finish crawling datasets from OpenTopography.
celery_worker_digital_twin  | [2025-03-05 05:05:39,461: WARNING/MainProcess] 2025-03-05 05:05:39 [newzealidar.datasets] INFO: Finish crawling datasets from OpenTopography.
celery_worker_digital_twin  | [2025-03-05 05:05:39,484: WARNING/MainProcess] --- Logging error ---
celery_worker_digital_twin  | [2025-03-05 05:05:39,485: WARNING/MainProcess] Traceback (most recent call last):
celery_worker_digital_twin  | [2025-03-05 05:05:39,486: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 451, in trace_task
celery_worker_digital_twin  |     R = retval = fun(*args, **kwargs)
celery_worker_digital_twin  |                  ^^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,486: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 734, in __protected_call__
celery_worker_digital_twin  |     return self.run(*args, **kwargs)
celery_worker_digital_twin  |            ^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,486: WARNING/MainProcess]   File "/app/src/tasks.py", line 142, in process_dem
celery_worker_digital_twin  |     process_hydro_dem.main(selected_polygon, **parameters)
celery_worker_digital_twin  | [2025-03-05 05:05:39,486: WARNING/MainProcess]   File "/app/src/flood_model/process_hydro_dem.py", line 121, in main
celery_worker_digital_twin  |     ensure_lidar_datasets_initialised()
celery_worker_digital_twin  | [2025-03-05 05:05:39,486: WARNING/MainProcess]   File "/app/src/flood_model/process_hydro_dem.py", line 49, in ensure_lidar_datasets_initialised
celery_worker_digital_twin  |     newzealidar.datasets.main()
celery_worker_digital_twin  | [2025-03-05 05:05:39,486: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/newzealidar/datasets.py", line 416, in main
celery_worker_digital_twin  |     rename_file()
celery_worker_digital_twin  | [2025-03-05 05:05:39,486: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/newzealidar/datasets.py", line 384, in rename_file
celery_worker_digital_twin  |     data_dir.mkdir(parents=True, exist_ok=True)
celery_worker_digital_twin  | [2025-03-05 05:05:39,487: WARNING/MainProcess]   File "/venv/lib/python3.11/pathlib.py", line 1116, in mkdir
celery_worker_digital_twin  |     os.mkdir(self, mode)
celery_worker_digital_twin  | [2025-03-05 05:05:39,487: WARNING/MainProcess] PermissionError: [Errno 13] Permission denied: '/stored_data/lidar'
celery_worker_digital_twin  | [2025-03-05 05:05:39,487: WARNING/MainProcess] 
celery_worker_digital_twin  | During handling of the above exception, another exception occurred:
celery_worker_digital_twin  | [2025-03-05 05:05:39,487: WARNING/MainProcess] Traceback (most recent call last):
celery_worker_digital_twin  | [2025-03-05 05:05:39,487: WARNING/MainProcess]   File "/venv/lib/python3.11/logging/__init__.py", line 1110, in emit
celery_worker_digital_twin  |     msg = self.format(record)
celery_worker_digital_twin  |           ^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,488: WARNING/MainProcess]   File "/venv/lib/python3.11/logging/__init__.py", line 953, in format
celery_worker_digital_twin  |     return fmt.format(record)
celery_worker_digital_twin  |            ^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,488: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/utils/log.py", line 146, in format
celery_worker_digital_twin  |     msg = super().format(record)
celery_worker_digital_twin  |           ^^^^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,488: WARNING/MainProcess]   File "/venv/lib/python3.11/logging/__init__.py", line 695, in format
celery_worker_digital_twin  |     record.exc_text = self.formatException(record.exc_info)
celery_worker_digital_twin  |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,488: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/utils/log.py", line 142, in formatException
celery_worker_digital_twin  |     r = super().formatException(ei)
celery_worker_digital_twin  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,488: WARNING/MainProcess]   File "/venv/lib/python3.11/logging/__init__.py", line 645, in formatException
celery_worker_digital_twin  |     traceback.print_exception(ei[0], ei[1], tb, None, sio)
celery_worker_digital_twin  | [2025-03-05 05:05:39,488: WARNING/MainProcess]   File "/venv/lib/python3.11/traceback.py", line 124, in print_exception
celery_worker_digital_twin  |     te = TracebackException(type(value), value, tb, limit=limit, compact=True)
celery_worker_digital_twin  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,488: WARNING/MainProcess]   File "/venv/lib/python3.11/traceback.py", line 688, in __init__
celery_worker_digital_twin  |     self.stack = StackSummary._extract_from_extended_frame_gen(
celery_worker_digital_twin  |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,489: WARNING/MainProcess]   File "/venv/lib/python3.11/traceback.py", line 416, in _extract_from_extended_frame_gen
celery_worker_digital_twin  |     for f, (lineno, end_lineno, colno, end_colno) in frame_gen:
celery_worker_digital_twin  | [2025-03-05 05:05:39,489: WARNING/MainProcess]   File "/venv/lib/python3.11/traceback.py", line 353, in _walk_tb_with_full_positions
celery_worker_digital_twin  |     positions = _get_code_position(tb.tb_frame.f_code, tb.tb_lasti)
celery_worker_digital_twin  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,489: WARNING/MainProcess]   File "/venv/lib/python3.11/traceback.py", line 366, in _get_code_position
celery_worker_digital_twin  |     positions_gen = code.co_positions()
celery_worker_digital_twin  |                     ^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,489: WARNING/MainProcess] AttributeError: '_Code' object has no attribute 'co_positions'
celery_worker_digital_twin  | [2025-03-05 05:05:39,489: WARNING/MainProcess] Call stack:
celery_worker_digital_twin  | [2025-03-05 05:05:39,490: WARNING/MainProcess]   File "/venv/lib/python3.11/threading.py", line 995, in _bootstrap
celery_worker_digital_twin  |     self._bootstrap_inner()
celery_worker_digital_twin  | [2025-03-05 05:05:39,490: WARNING/MainProcess]   File "/venv/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
celery_worker_digital_twin  |     self.run()
celery_worker_digital_twin  | [2025-03-05 05:05:39,490: WARNING/MainProcess]   File "/venv/lib/python3.11/threading.py", line 975, in run
celery_worker_digital_twin  |     self._target(*self._args, **self._kwargs)
celery_worker_digital_twin  | [2025-03-05 05:05:39,491: WARNING/MainProcess]   File "/venv/lib/python3.11/concurrent/futures/thread.py", line 83, in _worker
celery_worker_digital_twin  |     work_item.run()
celery_worker_digital_twin  | [2025-03-05 05:05:39,491: WARNING/MainProcess]   File "/venv/lib/python3.11/concurrent/futures/thread.py", line 58, in run
celery_worker_digital_twin  |     result = self.fn(*self.args, **self.kwargs)
celery_worker_digital_twin  | [2025-03-05 05:05:39,491: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/concurrency/base.py", line 29, in apply_target
celery_worker_digital_twin  |     ret = target(*args, **kwargs)
celery_worker_digital_twin  | [2025-03-05 05:05:39,491: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 649, in fast_trace_task
celery_worker_digital_twin  |     R, I, T, Rstr = tasks[task].__trace__(
celery_worker_digital_twin  | [2025-03-05 05:05:39,491: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 468, in trace_task
celery_worker_digital_twin  |     I, R, state, retval = on_error(task_request, exc, uuid)
celery_worker_digital_twin  | [2025-03-05 05:05:39,491: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 379, in on_error
celery_worker_digital_twin  |     R = I.handle_error_state(
celery_worker_digital_twin  | [2025-03-05 05:05:39,491: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 178, in handle_error_state
celery_worker_digital_twin  |     return {
celery_worker_digital_twin  | [2025-03-05 05:05:39,491: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 237, in handle_failure
celery_worker_digital_twin  |     self._log_error(task, req, einfo)
celery_worker_digital_twin  | [2025-03-05 05:05:39,491: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 265, in _log_error
celery_worker_digital_twin  |     logger.log(policy.severity, policy.format.strip(), context,
celery_worker_digital_twin  | [2025-03-05 05:05:39,491: WARNING/MainProcess] Message: 'Task %(name)s[%(id)s] %(description)s: %(exc)s'
celery_worker_digital_twin  | Arguments: {'hostname': 'celery@9c5163f19e5b', 'id': '122b1b89-984e-453d-aede-d469a9336093', 'name': 'src.tasks.process_dem', 'exc': "PermissionError(13, 'Permission denied')", 'traceback': 'Traceback (most recent call last):\n  File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 451, in trace_task\n    R = retval = fun(*args, **kwargs)\n                 ^^^^^^^^^^^^^^^^^^^^\n  File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 734, in __protected_call__\n    return self.run(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/app/src/tasks.py", line 142, in process_dem\n    process_hydro_dem.main(selected_polygon, **parameters)\n  File "/app/src/flood_model/process_hydro_dem.py", line 121, in main\n    ensure_lidar_datasets_initialised()\n  File "/app/src/flood_model/process_hydro_dem.py", line 49, in ensure_lidar_datasets_initialised\n    newzealidar.datasets.main()\n  File "/venv/lib/python3.11/site-packages/newzealidar/datasets.py", line 416, in main\n    rename_file()\n  File "/venv/lib/python3.11/site-packages/newzealidar/datasets.py", line 384, in rename_file\n    data_dir.mkdir(parents=True, exist_ok=True)\n  File "/venv/lib/python3.11/pathlib.py", line 1116, in mkdir\n    os.mkdir(self, mode)\nPermissionError: [Errno 13] Permission denied: \'/stored_data/lidar\'\n', 'args': "['POLYGON ((172.7131337872427 -43.39377030256972, 172.7131337872427 -43.38426113151589, 172.7001528519404 -43.38426113151589, 172.7001528519404 -43.39377030256972, 172.7131337872427 -43.39377030256972))']", 'kwargs': '{}', 'description': 'raised unexpected', 'internal': False}
celery_worker_digital_twin  | [2025-03-05 05:05:39,492: WARNING/MainProcess] --- Logging error ---
celery_worker_digital_twin  | [2025-03-05 05:05:39,493: WARNING/MainProcess] Traceback (most recent call last):
celery_worker_digital_twin  | [2025-03-05 05:05:39,494: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 451, in trace_task
celery_worker_digital_twin  |     R = retval = fun(*args, **kwargs)
celery_worker_digital_twin  |                  ^^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,494: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 734, in __protected_call__
celery_worker_digital_twin  |     return self.run(*args, **kwargs)
celery_worker_digital_twin  |            ^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,494: WARNING/MainProcess]   File "/app/src/tasks.py", line 142, in process_dem
celery_worker_digital_twin  |     process_hydro_dem.main(selected_polygon, **parameters)
celery_worker_digital_twin  | [2025-03-05 05:05:39,495: WARNING/MainProcess]   File "/app/src/flood_model/process_hydro_dem.py", line 121, in main
celery_worker_digital_twin  |     ensure_lidar_datasets_initialised()
celery_worker_digital_twin  | [2025-03-05 05:05:39,495: WARNING/MainProcess]   File "/app/src/flood_model/process_hydro_dem.py", line 49, in ensure_lidar_datasets_initialised
celery_worker_digital_twin  |     newzealidar.datasets.main()
celery_worker_digital_twin  | [2025-03-05 05:05:39,495: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/newzealidar/datasets.py", line 416, in main
celery_worker_digital_twin  |     rename_file()
celery_worker_digital_twin  | [2025-03-05 05:05:39,495: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/newzealidar/datasets.py", line 384, in rename_file
celery_worker_digital_twin  |     data_dir.mkdir(parents=True, exist_ok=True)
celery_worker_digital_twin  | [2025-03-05 05:05:39,495: WARNING/MainProcess]   File "/venv/lib/python3.11/pathlib.py", line 1116, in mkdir
celery_worker_digital_twin  |     os.mkdir(self, mode)
celery_worker_digital_twin  | [2025-03-05 05:05:39,495: WARNING/MainProcess] PermissionError: [Errno 13] Permission denied: '/stored_data/lidar'
celery_worker_digital_twin  | [2025-03-05 05:05:39,495: WARNING/MainProcess] 
celery_worker_digital_twin  | During handling of the above exception, another exception occurred:
celery_worker_digital_twin  | [2025-03-05 05:05:39,495: WARNING/MainProcess] Traceback (most recent call last):
celery_worker_digital_twin  | [2025-03-05 05:05:39,495: WARNING/MainProcess]   File "/venv/lib/python3.11/logging/__init__.py", line 1110, in emit
celery_worker_digital_twin  |     msg = self.format(record)
celery_worker_digital_twin  |           ^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,495: WARNING/MainProcess]   File "/venv/lib/python3.11/logging/__init__.py", line 953, in format
celery_worker_digital_twin  |     return fmt.format(record)
celery_worker_digital_twin  |            ^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,495: WARNING/MainProcess]   File "/venv/lib/python3.11/logging/__init__.py", line 695, in format
celery_worker_digital_twin  |     record.exc_text = self.formatException(record.exc_info)
celery_worker_digital_twin  |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/logging/__init__.py", line 645, in formatException
celery_worker_digital_twin  |     traceback.print_exception(ei[0], ei[1], tb, None, sio)
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/traceback.py", line 124, in print_exception
celery_worker_digital_twin  |     te = TracebackException(type(value), value, tb, limit=limit, compact=True)
celery_worker_digital_twin  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/traceback.py", line 688, in __init__
celery_worker_digital_twin  |     self.stack = StackSummary._extract_from_extended_frame_gen(
celery_worker_digital_twin  |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/traceback.py", line 416, in _extract_from_extended_frame_gen
celery_worker_digital_twin  |     for f, (lineno, end_lineno, colno, end_colno) in frame_gen:
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/traceback.py", line 353, in _walk_tb_with_full_positions
celery_worker_digital_twin  |     positions = _get_code_position(tb.tb_frame.f_code, tb.tb_lasti)
celery_worker_digital_twin  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/traceback.py", line 366, in _get_code_position
celery_worker_digital_twin  |     positions_gen = code.co_positions()
celery_worker_digital_twin  |                     ^^^^^^^^^^^^^^^^^
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess] AttributeError: '_Code' object has no attribute 'co_positions'
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess] Call stack:
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/threading.py", line 995, in _bootstrap
celery_worker_digital_twin  |     self._bootstrap_inner()
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/threading.py", line 1038, in _bootstrap_inner
celery_worker_digital_twin  |     self.run()
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/threading.py", line 975, in run
celery_worker_digital_twin  |     self._target(*self._args, **self._kwargs)
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/concurrent/futures/thread.py", line 83, in _worker
celery_worker_digital_twin  |     work_item.run()
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/concurrent/futures/thread.py", line 58, in run
celery_worker_digital_twin  |     result = self.fn(*self.args, **self.kwargs)
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/concurrency/base.py", line 29, in apply_target
celery_worker_digital_twin  |     ret = target(*args, **kwargs)
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 649, in fast_trace_task
celery_worker_digital_twin  |     R, I, T, Rstr = tasks[task].__trace__(
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 468, in trace_task
celery_worker_digital_twin  |     I, R, state, retval = on_error(task_request, exc, uuid)
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 379, in on_error
celery_worker_digital_twin  |     R = I.handle_error_state(
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 178, in handle_error_state
celery_worker_digital_twin  |     return {
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 237, in handle_failure
celery_worker_digital_twin  |     self._log_error(task, req, einfo)
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess]   File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 265, in _log_error
celery_worker_digital_twin  |     logger.log(policy.severity, policy.format.strip(), context,
celery_worker_digital_twin  | [2025-03-05 05:05:39,496: WARNING/MainProcess] Message: 'Task %(name)s[%(id)s] %(description)s: %(exc)s'
celery_worker_digital_twin  | Arguments: {'hostname': 'celery@9c5163f19e5b', 'id': '122b1b89-984e-453d-aede-d469a9336093', 'name': 'src.tasks.process_dem', 'exc': "PermissionError(13, 'Permission denied')", 'traceback': 'Traceback (most recent call last):\n  File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 451, in trace_task\n    R = retval = fun(*args, **kwargs)\n                 ^^^^^^^^^^^^^^^^^^^^\n  File "/venv/lib/python3.11/site-packages/celery/app/trace.py", line 734, in __protected_call__\n    return self.run(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/app/src/tasks.py", line 142, in process_dem\n    process_hydro_dem.main(selected_polygon, **parameters)\n  File "/app/src/flood_model/process_hydro_dem.py", line 121, in main\n    ensure_lidar_datasets_initialised()\n  File "/app/src/flood_model/process_hydro_dem.py", line 49, in ensure_lidar_datasets_initialised\n    newzealidar.datasets.main()\n  File "/venv/lib/python3.11/site-packages/newzealidar/datasets.py", line 416, in main\n    rename_file()\n  File "/venv/lib/python3.11/site-packages/newzealidar/datasets.py", line 384, in rename_file\n    data_dir.mkdir(parents=True, exist_ok=True)\n  File "/venv/lib/python3.11/pathlib.py", line 1116, in mkdir\n    os.mkdir(self, mode)\nPermissionError: [Errno 13] Permission denied: \'/stored_data/lidar\'\n', 'args': "['POLYGON ((172.7131337872427 -43.39377030256972, 172.7131337872427 -43.38426113151589, 172.7001528519404 -43.38426113151589, 172.7001528519404 -43.39377030256972, 172.7131337872427 -43.39377030256972))']", 'kwargs': '{}', 'description': 'raised unexpected', 'internal': False}
```

**Comment by LukeParky** [comment]
Hi @pritamd47, to confirm this problem, could you please let me know the OS you were running this in? I think from our call yesterday it was running on WSL so you can get the OS through `cat /etc/os-release`.

**Comment by LukeParky** [comment]
Okay, I can confirm this happens in other OSs, its much more closely related to #344 than the issue that you originally opened. I believe this issue (Increase explanations for POSTGRES_PASSWORD .env) is fixed, but I will reopen #344.

**Comment by pritamd47** [comment]
Hi @LukeParky, the OS I was running was Ubuntu. I see that you have reopened the issue #344. I will follow that thread for updates on the fix. Let me know when you're able to push fix, and I will rerun the application.

**Comment by LukeParky** [comment]
> Hi [@LukeParky](https://github.com/LukeParky), the OS I was running was Ubuntu. I see that you have reopened the issue [#344](https://github.com/GeospatialResearch/Digital-Twins/issues/344). I will follow that thread for updates on the fix. Let me know when you're able to push fix, and I will rerun the application.

Hi @pritamd47, could you let me know the exact version of Ubuntu you were running? I want to complete some more testing on that version specifically

**Comment by pritamd47** [comment]
Hi @LukeParky. Please see the latest logs. The celery worker is getting killed, and I am facing the same issue as before "Could not reach backend. Please refresh the page to try again."
 
![Image](https://github.com/user-attachments/assets/1f13bfe5-0df6-4bf1-abb2-8055a8134e93)

**Comment by LukeParky** [comment]
I have seen this happen when the computer runs out of memory, do you think this could be the case?

**Comment by pritamd47** [comment]
Yes, that was likely the case. I cleared the memory and ran again after which it ran smoothly. However, I am still getting the `permission_denied` error #344:

![Image](https://github.com/user-attachments/assets/fec17726-6a76-403c-b32c-05e65e58fa7e)

I updated the repository using `git pull` before running any of the docker commands.

**Comment by LukeParky** [comment]
> I updated the repository using `git pull` before running any of the docker commands.


Can you confirm that the docker commands you ran were as stated in #344, I am not sure how to tag you in that issue.

You need new volumes because of the bug so you need to run:
`git pull && docker compose pull && docker compose down -v && docker compose up -d && docker compose logs -f backend celery_worker`

Can you please confirm this? Because if the error still occurs after running this then there is something else major causing issues that I cannot replicate.

**Comment by pritamd47** [comment]
Hi Luke, yes these are the commands that I used for starting up the server.

**Comment by LukeParky** [comment]
@pritamd47 To help me solve this problem could you please give me the output of the following two commands:

`docker compose images`

`docker volume inspect digital-twins_stored_data`
Thank you

**Comment by pritamd47** [comment]
Output of `docker compose images` ![Image](https://github.com/user-attachments/assets/8db15552-927e-430e-a2a3-c9c2439a762e)

Output of `docker volume inspect digital-twins_stored_data` 
![Image](https://github.com/user-attachments/assets/810a4715-c636-4390-9707-a48967b0f916)

**Comment by pritamd47** [comment]
Hi @LukeParky, any updates on the issue?

**Comment by LukeParky** [comment]
Hi @pritamd47 I have moved discussions onto the other issue, but I can't tag you in it until you have commented there.

Please see my comment https://github.com/GeospatialResearch/Digital-Twins/issues/344#issuecomment-2885577677
